### PR TITLE
Protobuf definition for Optimint types

### PR DESCRIPTION
**Pull request by tzdybal**
This PR changes serialization from GOB to protobuf.
It's not targeting `main` branch, because it depends on some refactorings introduced in `tzdybal/payformessage`.

### CODE REVIEW

File: proto/tendermint/consensus/types.proto [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+// NewRoundStep is sent for every step taken in the ConsensusState.
+// For every height/round/step transition
+message NewRoundStep {
+  int64  height                   = 1;
+  int32  round                    = 2;
+  uint32 step                     = 3;
+  int64  seconds_since_start_time = 4;
+  int32  last_commit_round        = 5;
+}
```
Comment (liamsi) [inline proto/tendermint/consensus/types.proto:18]: Does this go through the ABCI boundary? I don't think rounds are necessary in optimint.

File: proto/tendermint/consensus/types.proto [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    NewValidBlock new_valid_block = 2;
+    Proposal      proposal        = 3;
+    ProposalPOL   proposal_pol    = 4;
+    BlockPart     block_part      = 5;
+    Vote          vote            = 6;
+    HasVote       has_vote        = 7;
+    VoteSetMaj23  vote_set_maj23  = 8;
+    VoteSetBits   vote_set_bits   = 9;
+  }
+}
```
Comment (liamsi) [inline proto/tendermint/consensus/types.proto:92]: All these messages should not need to be part of optimint as they are tendermint consensus specific.

File: proto/tendermint/consensus/wal.proto [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    TimeoutInfo                          timeout_info           = 3;
+    EndHeight                            end_height             = 4;
+  }
+}
+
+// TimedWALMessage wraps WALMessage and adds Time for debugging purposes.
+message TimedWALMessage {
+  google.protobuf.Timestamp time = 1 [(gogoproto.nullable) = false, (gogoproto.stdtime) = true];
+  WALMessage                msg  = 2;
+}
```
Comment (liamsi) [inline proto/tendermint/consensus/wal.proto:46]: This is also tendermint specific.

**Review by liamsi — CHANGES_REQUESTED** [review]
This should only contain the protobuf types that are necessary because either
1) they are core types in optimint or
2) they are necessary because of ABCI

File: proto/tendermint/abci/types.proto [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+// This file is copied from http://github.com/tendermint/abci
+// NOTE: When using custom types, mind the warnings.
+// https://github.com/gogo/protobuf/blob/master/custom_types.md#warnings-and-issues
+
+// Validator
+message Validator {
+  bytes address = 1;  // The first 20 bytes of SHA256(public key)
+  // PubKey pub_key = 2 [(gogoproto.nullable)=false];
+  int64 power = 3;  // The voting power
+}
```
Comment (liamsi) [inline proto/tendermint/abci/types.proto:20]: We might want to change that to Aggregator. But we can think about this further and do this in a followup PR.

**Review by liamsi — APPROVED** [review]
LGTM

File: da/lazyledger/lazyledger_test.go [hunk 5]
```diff
@@ -43,6 +43,7 @@ func TestConfiguration(t *testing.T) {
 }
 
 func TestSubmission(t *testing.T) {
+	t.Skip("unfinished test/implementation")
```
[the remark this answers is not in the document]
Comment (tzdybal) [reply da/lazyledger/lazyledger_test.go:46]: This PR is targetting `tzdybal/payformessage` and the test is from that base branch. It's not working yet, and is under development there. It's not related to protobuf changes anyways.

**Pull request merged**